war/src/main/java/io/lumeer/engine/controller/UserGroupFacade.java

```diff
@@ -0,0 +1,261 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.lumeer.engine.controller;
+
+import io.lumeer.engine.annotation.SystemDataStorage;
+import io.lumeer.engine.api.LumeerConst;
+import io.lumeer.engine.api.LumeerConst.Group;
+import io.lumeer.engine.api.LumeerConst.UserGroup;
+import io.lumeer.engine.api.data.DataDocument;
+import io.lumeer.engine.api.data.DataFilter;
+import io.lumeer.engine.api.data.DataStorage;
+import io.lumeer.engine.api.data.DataStorageDialect;
+import io.lumeer.engine.api.exception.UserAlreadyExistsException;
+import io.lumeer.engine.util.ErrorMessageBuilder;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.context.SessionScoped;
+import javax.inject.Inject;
+
+/**
+ * Stores users and their groups
+ */
+@SessionScoped
+public class UserGroupFacade implements Serializable {
+
+   @Inject
+   @SystemDataStorage
+   private DataStorage dataStorage;
+
+   @Inject
+   private DataStorageDialect dataStorageDialect;
+
+   /**
+    * Adds user to organization with one or more groups.
+    *
+    * @param organization
+    *       Id of the organization.
+    * @param user
+    *       User identificator.
+    * @param groups
+    *       One or more group names.
+    * @throws UserAlreadyExistsException
+    *       When user already exists in organization.
+    */
+   public void addUser(final String organization, final String user, final String... groups) throws UserAlreadyExistsException {
+      if (readUser(organization, user) != null) {
+         throw new UserAlreadyExistsException(ErrorMessageBuilder.userAlreadyExistsInOrganizationString(user, organization));
+      }
+      DataDocument document = new DataDocument(UserGroup.ATTR_USERS_USER, user)
+            .append(UserGroup.ATTR_USERS_GROUPS, Arrays.asList(groups));
+      dataStorage.addItemToArray(UserGroup.COLLECTION_NAME, organizationFilter(organization), UserGroup.ATTR_USERS, document);
+   }
+
+   /**
+    * Adds one or more groups to organization
+    *
+    * @param organization
+    *       Id of the organization.
+    * @param groups
+    *       The one or more group names.
+    */
+   public void addGroup(final String organization, final String... groups) {
```

Comment:
I think all these methods where there is `String... groups` parameter should be called `addGroups` etc. (plural form).